backend/corpora/dataset_submissions/app.py

```diff
@@ -0,0 +1,114 @@
+import logging
+import os
+import re
+from typing import Tuple, Optional
+from urllib.parse import unquote_plus
+
+from sqlalchemy.orm import Session
+
+from backend.corpora.common.entities import Collection
+from backend.corpora.common.upload_sfn import upload
+from backend.corpora.common.utils.db_session import db_session_manager
+from backend.corpora.common.utils.exceptions import CorporaException
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+UUID_REGEX = "[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}"
+
+
+def dataset_submissions_handler(s3_event: dict, context) -> None:
+    """
+    Lambda function invoked when a dataset is uploaded to s3://cellxgene-dataset-submissions-{dev,staging,prod}
+    :param s3_event: Lambda's event object
+    :param context: Lambda's context object
+    :return:
+    """
+    logger.info(f"@@@@@ {s3_event=}")
+    logger.info(f"@@@@@ {os.environ.get('REMOTE_DEV_PREFIX', '')=}")
+
+    # s3://cellxgene-curation-submissions-{dev,staging,prod}/<collection_id>/<curator_tag>
```

Comment:
I would change this comment to reflect the fact that prod won't have the env appended. The prod bucket will be `cellxgene-curation-submissions`.